tensorzero-core/src/client/mod.rs

```diff
@@ -380,6 +398,65 @@ impl ClientBuilder {
                     last_body: Default::default(),
                 })
             }
+            ClientBuilderMode::FromComponents {
+                config,
+                clickhouse_connection_info,
+                postgres_connection_info,
+                http_client,
+                timeout,
+            } => {
+                // Validation: Batch writes are not supported in embedded mode
+                if config.gateway.observability.batch_writes.enabled
```

Comment:
Would you mind extracting this logic out to a common function, to make sure our checks stay in sync between the various ways of constructing an embedded client?